### PR TITLE
Allow processor to be initialized with Sass config options that are passed to Sass engine initialization

### PR DESCRIPTION
**Pull request by agrobbin**
Required to fix rails/sass-rails#358.

I wasn't sure whether to add the same option to `SasscProcessor`, but can definitely to add it to this PR if you'd like!

/cc @rafaelfranca

### CODE REVIEW

**Comment by rafaelfranca** [comment]
Could you see if that is useful for SassCProcessor too?

**Comment by agrobbin** [comment]
@rafaelfranca it looks like sassc-rails isn't using the sprockets `SasscProcessor` class yet, it's still using Sprockets 2/3 engines rather than Sprockets 3/4 transformers. Definitely seems worth adding it for the time when `sassc-rails` is updated to use the built-in `SasscProcessor` class.

**Comment by rafaelfranca** [comment]
Our plan is only have sass-rails working for sassc and sass, so if Sassc::Engine can be configured in the same way that Sass::Engine, so we need to change the SassCProcessor even that sassc-rails is not using it yet.

**Comment by agrobbin** [comment]
Done!

File: lib/sprockets/sass_processor.rb [hunk 1]
```diff
@@ -68,9 +69,9 @@ def call(input)
           environment: input[:environment],
           dependencies: context.metadata[:dependencies]
         }
-      }
+      }.merge!(@sass_config)
```
Comment (lucasmazza) [inline lib/sprockets/sass_processor.rb]: Shouldn't we do some sort of deep merge here? The `load_paths` entry gets replaced by an empty `Array` from the defaults of [`sass-rails`](https://github.com/rails/sass-rails/blob/a11d4c7989a2b3e82b780e4c363c7e5d28c4b739/lib/sass/rails/railtie.rb#L19-L20) and it seems to be breaking `@import` statements.

Comment (rafaelfranca) [inline lib/sprockets/sass_processor.rb]: :+1:

**Pull request merged**